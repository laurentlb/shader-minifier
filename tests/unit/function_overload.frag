int sum0() { return 0; }

int sum1(int x) { return sum0() + x; }

int sum2(int x, int y) { return sum1(x) + y; }

int sum3(int x, int y, int z) { return sum2(x, y) + z; }

int sum4(int x, int y, int z, int w) { return sum3(x, y, z) + w; }

float sum5(vec2 v2) { return length(v2); }

float sum6(vec3 v3) { return length(v3); }

float sum7(int x, vec2 v2) { return length(v2) + float(x); }

float sum8(float x, vec2 v2) { return length(v2) + x; }

int mult1(int x) { return x; }

int mult2(int x, int y) { return mult1(x) * y; }

int mult3(int x, int y, int z) { return mult2(x, y) * z; }

void mult7(const int x, vec2 v2) {}

int mult8(float x, in vec2 v2) { return 0; }
