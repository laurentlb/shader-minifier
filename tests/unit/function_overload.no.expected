// Generated with  (https://github.com/laurentlb/Shader_Minifier/)
#ifndef FUNCTION_OVERLOAD_NO_EXPECTED_
# define FUNCTION_OVERLOAD_NO_EXPECTED_

const char *function_overload_frag =
 "int h()"
 "{"
   "return 0;"
 "}"
 "int l(int l)"
 "{"
   "return h()+l;"
 "}"
 "int i(int i,int H)"
 "{"
   "return l(i)+H;"
 "}"
 "int t(int l,int H,int I)"
 "{"
   "return i(l,H)+I;"
 "}"
 "int N(int i,int l,int H,int I)"
 "{"
   "return t(i,l,H)+I;"
 "}"
 "float M(vec2 i)"
 "{"
   "return length(i);"
 "}"
 "float L(vec3 i)"
 "{"
   "return length(i);"
 "}"
 "float K(int i,vec2 l)"
 "{"
   "return length(l)+float(i);"
 "}"
 "float J(float l,vec2 i)"
 "{"
   "return length(i)+l;"
 "}"
 "int I(int i)"
 "{"
   "return i;"
 "}"
 "int H(int i,int l)"
 "{"
   "return I(i)*l;"
 "}"
 "int G(int i,int l,int I)"
 "{"
   "return H(i,l)*I;"
 "}"
 "void F(const int i,vec2 l){}"
 "int E(float i,vec2 l)"
 "{"
   "return 0;"
 "}";

#endif // FUNCTION_OVERLOAD_NO_EXPECTED_
